File: src/main/java/com/example/dgsexample/fetcher/CompanyService.java
package com.example.dgsexample.fetcher;

import com.example.dgsexample.model.Company;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CompanyService {
    @Autowired
    CompanyRepository companyRepository;

    public List<Company> findByCompanyName(String companyName) {
        return companyRepository.findByCompanyName(companyName);
    }
}

File: src/main/java/com/example/dgsexample/fetcher/CompanyService.kt
package com.example.dgsexample.fetcher

import com.example.dgsexample.model.Company
import org.springframework.stereotype.Service

@Service
class CompanyService(val companyRepository: CompanyRepository) {

    fun findByCompanyName(companyName: String?): List<Company> {
        if (companyName != null) {
            return companyRepository.findByCompanyName(companyName);
        }

        return companyRepository.findAll();
    }
}